rlclientlib/model_mgmt/restapi_data_transport.cc

```diff
@@ -51,8 +51,8 @@ namespace reinforcement_learning { namespace model_management {
     // Build request URI and start the request.
     auto request_task = _httpcli->request(request).then([&](http_response response) {
       if (response.status_code() != 200){
-        //if the call using HEAD fails with 404 try with GET only once and return the results of GET request call
-        if (response.status_code() == 404 && _retry_get_data){
+        //if the call using HEAD fails try with GET only once and return the results of GET request call
+        if (_retry_get_data){

```

Comment:
what is the motivation behind removing this 404 check?

Reply:
there are two ways we are getting the model. One is using the azure blob uri and the other is using model API in front end. azure blob uri support HEAD request where as model API doesn't at the moment. So initially when we try with head using model API it throws 404 but when tested using local host it's throwing 405. So I thought irrespective of error code we give another try with GET instead of HEAD when the call with HEAD fails